src/Microsoft.AspNet.Mvc.Core/Formatters/MediaTypeComponent.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Internal;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNet.Mvc.Formatters
+{
+    /// <summary>
+    /// Represents a component within a media type.
```

Comment:
I can be more verbose and explain that the components are type, subtype and the parameters of the media type. Is that what you want?